tests/schema/tests.py

```diff
@@ -3175,3 +3175,193 @@ class Meta:
+        self.assertEqual(table_description[1].collation, collation)
+
+        field_without_collation = TextField(db_collation=None)
+        field_without_collation.set_attributes_from_name('field')
+        with connection.schema_editor() as editor:
+            editor.alter_field(
+                CollationTest, CollationTest._meta.get_field('field'), field_without_collation, strict=True
+            )
+        with connection.cursor() as cursor:
+            table_description = connection.introspection.get_table_description(cursor, CollationTest._meta.db_table)
+        if connection.vendor == 'mysql':
+            # MySQL gives the collation of the table if there's no collation set.
+            self.assertNotEqual(table_description[1].collation, collation)
+        elif connection.vendor == 'postgresql':
+            self.assertEqual(table_description[1].collation, 'default')
+        else:
+            self.assertIsNone(table_description[1].collation)
+
+    @isolate_apps('schema')
+    def test_collations_none_to_collated(self):
+        if connection.vendor == 'oracle':
+            with connection.cursor() as cursor:
+                if connection.introspection.get_max_string_size(cursor) == 'STANDARD':
+                    self.skipTest('Oracle only supports collations with extended max_string_size')
+
+        collation = {
+            'mysql': 'latin1_general_cs',
+            'oracle': 'BINARY',
+            'postgresql': 'en_US',
+            'sqlite': 'binary',
+        }[connection.vendor]
+
+        class CollationTest(Model):
+            field = CharField(max_length=255)
+
+            class Meta:
+                app_label = 'schema'
+
+        self.isolated_local_models = [CollationTest]
+        with connection.schema_editor() as editor:
+            editor.create_model(CollationTest)
+
+        new_field = CharField(max_length=255, db_collation=collation)
+        new_field.set_attributes_from_name("field")
+        with connection.schema_editor() as editor:
+            editor.alter_field(CollationTest, CollationTest._meta.get_field('field'), new_field, strict=True)
+        with connection.cursor() as cursor:
+            table_description = connection.introspection.get_table_description(cursor, CollationTest._meta.db_table)
+        self.assertEqual(table_description[1].collation, collation)
+
+    @isolate_apps('schema')
+    @unittest.skipIf(
+        connection.vendor == 'postgresql' and not connection.features.is_postgresql_12,
+        'PG < 12 has no non-deterministic collations'
+    )
+    def test_ci_cs_collations(self):
+        if connection.vendor == 'oracle':
+            with connection.cursor() as cursor:
+                if connection.introspection.get_max_string_size(cursor) == 'STANDARD':
+                    self.skipTest('Oracle only supports collations with extended max_string_size')
```

Comment:
Wonder if we could at least give the suite a run with `MAX_STRING_SIZE=EXTENDED` to ensure it works appropriately. I assume you came up with the `BINARY_CI` collation by running the tests locally @knyghty?

Reply:
I was just about to make this comment. Yeah, I have the tests running and passing locally with `MAX_STRING_SIZE=EXTENDED`. We should set up Jenkins for this, but the process is predictably Oracular: https://docs.oracle.com/en/database/oracle/oracle-database/18/refrn/MAX_STRING_SIZE.html#GUID-D424D23B-0933-425F-BC69-9C0E6724693C